qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp

```diff
@@ -0,0 +1,414 @@
+  batchAlgoRunner->executeBatchAsync();
+
+  return outputGroupName;
+}
+
+std::pair<std::string, std::string> IETModel::createGrouping(const IETGroupingData &groupingData,
+                                                             const IETConversionData &conversionData) {
+  auto groupType = groupingData.getGroupingType();
+
+  if (groupType == IETGroupingType::FILE) {
+    return std::make_pair(IETGroupingType::FILE, groupingData.getGroupingMapFile());
+  } else if (groupType == IETGroupingType::GROUPS) {
+    std::string groupingString = getDetectorGroupingString(conversionData.getSpectraMin(),
+                                                           conversionData.getSpectraMax(), groupingData.getNGroups());
+    return std::make_pair(IETGroupingType::CUSTOM, groupingString);
+  } else if (groupType == IETGroupingType::DEFAULT) {
+    return std::make_pair(IETGroupingType::IPF, "");
+  } else if (groupType == IETGroupingType::CUSTOM)
+    return std::make_pair(IETGroupingType::CUSTOM, groupingData.getCustomGroups());
+  else {
+    // Catch All and Individual
+    return std::make_pair(groupType, "");
+  }
+}
+
+std::string IETModel::getDetectorGroupingString(int spectraMin, int spectraMax, int nGroups) {
+  const unsigned int nSpectra = 1 + spectraMax - spectraMin;
+  return createDetectorGroupingString(static_cast<std::size_t>(nSpectra), static_cast<std::size_t>(nGroups),
+                                      static_cast<std::size_t>(spectraMin));
+}
+
+std::vector<std::string> IETModel::validatePlotData(IETPlotData plotParams) {
+  std::vector<std::string> errors;
+
+  const std::string inputFiles = plotParams.getInputData().getInputFiles();
+  if (inputFiles.empty()) {
+    errors.push_back("You must select a run file.");
+  }
+
+  IETDataValidator validator;
+
+  bool isRunFileValid = !inputFiles.empty();
+  std::string firstFileName = inputFiles.substr(0, inputFiles.find(','));
+
+  std::string conversionError = validator.validateConversionData(plotParams.getConversionData());
+  if (!conversionError.empty()) {
+    errors.push_back(conversionError);
+  }
+
+  std::vector<std::string> backgroundErrors = validator.validateBackgroundData(
+      plotParams.getBackgroundData(), plotParams.getConversionData(), firstFileName, isRunFileValid);
+  errors.insert(errors.end(), backgroundErrors.begin(), backgroundErrors.end());
+
+  errors.erase(std::remove(errors.begin(), errors.end(), ""), errors.end());
+
+  return errors;
+}
+
+void IETModel::plotRawFile(MantidQt::API::BatchAlgorithmRunner *batchAlgoRunner, InstrumentData instData,
+                           IETPlotData plotParams) {
+  using Mantid::specnum_t;
+
+  const std::string inputFiles = plotParams.getInputData().getInputFiles();
+
+  int spectraMin = plotParams.getConversionData().getSpectraMin();
+  int spectraMax = plotParams.getConversionData().getSpectraMax();
+
+  std::string rawFile = inputFiles.substr(0, inputFiles.find(',')); // getting the name of the first file
```

Comment:
why does finding the "," help?

Reply:
This assumes the input files string format is comma separated so it truncates the string until the first "," to get the name of the first file.

Reply:
ah ok, its splitting on files. I thought it was getting the name from the full filename so there shouldnt be a comma. 

This now makes sense